tests/system/src/test/java/test/robot/javafx/scene/TextFlow_TextLayout_Test.java

```diff
@@ -0,0 +1,382 @@
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.robot.javafx.scene;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.function.Consumer;
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.geometry.Rectangle2D;
+import javafx.scene.Node;
+import javafx.scene.Scene;
+import javafx.scene.robot.Robot;
+import javafx.scene.text.CaretInfo;
+import javafx.scene.text.Font;
+import javafx.scene.text.LayoutInfo;
+import javafx.scene.text.Text;
+import javafx.scene.text.TextFlow;
+import javafx.scene.text.TextLineInfo;
+import javafx.stage.Stage;
+import javafx.stage.StageStyle;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import test.util.Util;
+
+/*
+ * Tests new LayoutInfo API in the TextFlow.
+ */
+public class TextFlow_TextLayout_Test {
+    private static final int WIDTH = 500;
+    private static final int HEIGHT = 500;
+    private static final int FONT_SIZE = 24;
+    private static final double LINE_SPACING = 33;
+    private static final double EPS = 0.1;
+    private static final CountDownLatch startupLatch = new CountDownLatch(1);
+    private static Robot robot;
+    private static TextFlow textFlow;
+    private static volatile Stage stage;
+    private static volatile Scene scene;
+
+    // testing caret info
+    @Test
+    public void testCaretInfo() {
+        setText("__________\n", "______\n", "_\n");
+        waitForIdle();
+        LayoutInfo la = textFlow.getLayoutInfo();
+
+        CaretInfo ci = la.caretInfoAt(0, true);
+
+        // caret is one line
+        assertEquals(ci.getSegmentCount(), 1);
```

Comment:
Expected and actual are reversed, here and a few others below (most are OK).

Reply:
in the other test as well...  good catch!